src/sonic-yang-models/tests/yang_model_tests/yangTest.json

```diff
@@ -812,6 +812,48 @@
         }
     },
 
+    "DEVICE_METADATA_CORRECT_BUFFER_MODEL_PATTERN": {
+        "sonic-device_metadata:sonic-device_metadata": {
+            "sonic-device_metadata:DEVICE_METADATA": {
+                "sonic-device_metadata:localhost": {
+                "docker_routing_config_mode": "separated",
+                "platform": "x86_64-mlnx_msn4700-r0",
+                "hostname": "DUT-ASW",
+                "bgp_asn": "65000",
+                "buffer_model": "dynamic"
+                }
+            }
+        }
+    },
+
+    "DEVICE_METADATA_CORRECT_BUFFER_MODEL_PATTERN2": {
+        "sonic-device_metadata:sonic-device_metadata": {
+            "sonic-device_metadata:DEVICE_METADATA": {
+                "sonic-device_metadata:localhost": {
+                "docker_routing_config_mode": "separated",
+                "platform": "x86_64-mlnx_msn4700-r0",
+                "hostname": "DUT-ASW",
+                "bgp_asn": "65000",
+                "buffer_model": "traditional"
+                }
+            }
+        }
+    },
+
+    "DEVICE_METADATA_INCORRECT_BUFFER_MODEL_PATTERN": {
+        "sonic-device_metadata:sonic-device_metadata": {
+            "sonic-device_metadata:DEVICE_METADATA": {
+                "sonic-device_metadata:localhost": {
+                "docker_routing_config_mode": "separated",
+                "platform": "x86_64-mlnx_msn4700-r0",
+                "hostname": "DUT-ASW",
+                "bgp_asn": "65000",
+                "buffer_model": "separated"
```

Comment:
It will be great if the value itself can be "incorrect_pattern" instead of "separated". 
ping me after that and I will approve, thx a lot.

Reply:
@praveen-li, done